ta4j-core/src/test/java/org/ta4j/core/analysis/criteria/RewardRiskRatioCriterionTest.java

```diff
@@ -68,30 +68,28 @@ public void rewardRiskRatioCriterionOnlyWithGain() {
         TradingRecord tradingRecord = new BaseTradingRecord(
                 Order.buyAt(0, series), Order.sellAt(1, series),
                 Order.buyAt(2, series), Order.sellAt(5, series));
-        assertTrue(Double.isInfinite(rrc.calculate(series, tradingRecord)));
+        assertTrue(rrc.calculate(series, tradingRecord).isNaN());
     }
 
     @Test
     public void rewardRiskRatioCriterionWithNoTrades() {
         MockTimeSeries series = new MockTimeSeries(numFunction, 1, 2, 3, 6, 8, 20, 3);
-        assertTrue(Double.isInfinite(rrc.calculate(series, new BaseTradingRecord())));
+        assertTrue(rrc.calculate(series, new BaseTradingRecord()).isNaN());
     }
-    
+
     @Test
     public void withOneTrade() {
         MockTimeSeries series = new MockTimeSeries(numFunction, 100, 95, 95, 100, 90, 95, 80, 120);
         Trade trade = new Trade(Order.buyAt(0, series), Order.sellAt(1, series));
 
-
-
         RewardRiskRatioCriterion ratioCriterion = new RewardRiskRatioCriterion();
-        assertEquals((95d/100) / ((1d - 0.95d)), TestUtils.GENERAL_OFFSET, ratioCriterion.calculate(series, trade));
+        assertEquals((95d/100) / ((1d - 0.95d)), ratioCriterion.calculate(series, trade).doubleValue(), GENERAL_OFFSET);
     }
 
     @Test
     public void betterThan() {
         AnalysisCriterion criterion = new RewardRiskRatioCriterion();
```

Comment:
This criterion does not extend `AbstractCriterionTest`.  OOS?

Reply:
Same as above, doesn't work, basically we have two options:

a)

```java
    @Before
    public void setUp() {
        this.rrc = new RewardRiskRatioCriterion();
    }

    @Test
    public void betterThan() {
        assertTrue(rrc.betterThan(numOf(3.5), numOf(2.2)));
        assertFalse(rrc.betterThan(numOf(1.5), numOf(2.7)));
    }
``

b)

```java
    public RewardRiskRatioCriterionTest(Function<Number, Num> numFunction) {
        super((params) -> new RewardRiskRatioCriterion(), numFunction);
    }

    @Test
    public void betterThan() {
        assertTrue(getCriterion().betterThan(numOf(3.5), numOf(2.2)));
        assertFalse(getCriterion().betterThan(numOf(1.5), numOf(2.7)));
    }
```

`A` is good enough for all tests minus 1, as only 1 of the Criterion takes arguments, and I think `A` is a little easier to read/understand.

If we do that we should change it on all other places in the tests as well, as there are a lot of places where we create an instance of the creation, so to keep it consistent I propose to change it in *all* the places.

Reply:
B.  Facilitates new test case creation by copy-and-paste from existing examples.

Reply:
I'm still recommending B here.  Regardless, #214 will force option B so we can call this OOS and allow it.